src/data-access/FetchSearchEntityRepository.ts

```diff
@@ -0,0 +1,48 @@
+import SearchEntityRepository from '@/data-access/SearchEntityRepository';
+import SearchResult from '@/data-access/SearchResult';
+import TechnicalProblem from '@/data-access/errors/TechnicalProblem';
+
+export default class FetchSearchEntityRepository implements SearchEntityRepository {
+	private readonly forLanguageCode: string;
+	private readonly endpoint: string;
+
+	public constructor( languageCode: string, endpoint: string ) {
+		this.forLanguageCode = languageCode;
+		this.endpoint = endpoint;
+	}
+
+	public async searchProperties( searchString: string, limit?: number, offset?: number ):
+	Promise<SearchResult[]> {
+		const params: { [key: string]: string|number } = {
+			action: 'wbsearchentities',
+			search: searchString,
+			language: this.forLanguageCode,
+			type: 'property',
+			format: 'json',
+			formatversion: '2',
+			errorformat: 'plaintext',
+			origin: '*'
+		};
+		if ( limit ) {
+			params.limit = limit;
+		}
+		if ( offset ) {
+			params.offset = offset;
+		}
+
+		const query = Object.entries( params )
+			.map( ( entry ) => entry.join( '=' ) )
+			.join( '&' );
+		const url = `${this.endpoint}?${query}`;
```

Comment:
This will be prone to reflective XSS attacks. You can use URL object https://javascript.info/url instead.

Reply:
How? This URL never makes its way into HTML.

But you are probably right, using the URL objects should result in nicer code. I'll look for an [IE polyfill](https://caniuse.com/?search=URL)

Reply:
> How? This URL never makes its way into HTML.
It can break out and run custom javascript here.
> 
> But you are probably right, using the URL objects should result in nicer code. I'll look for an [IE polyfill](https://caniuse.com/?search=URL)
Half of the things we use is not available in IE11 (most of ES6), I thought we do babbel for that (We should!)

Reply:
The polyfill seems to come with core-js, so we should be good. :heavy_check_mark:

Reply:
>> How? This URL never makes its way into HTML.
> It can break out and run custom javascript here.

Still not seeing how. Could you explain? This url is used in a fetch call. The only way that I see, that this could maybe used for XSS would be if we were to display the unescaped URL in an error message.

Reply:
What I meant was when you concat user input string, it can break out and run arbitrary javascript. I can find some materials in the internet.